data-modeling.md

```diff
@@ -0,0 +1,64 @@
+---
+layout: default
+title: Data Modeling
+nav_order: 4
+has_toc: true
+---
+
+# Data Modeling
+
+With Dojo's Data Modeling tool, users can load features from NetCDF files into a visual graphing flow, perform actions on those features, and generate new datasets from the results.
+
+## Selecting NetCDFs
+
+When you initially land on the Data Modeling page, you'll be prompted to selected registered NetCDFs. If you have a NetCDF that you'd like to use that isn't yet registered in Dojo, follow the steps in [Data Registration](./data-registration) to register it. It will then appear in the list of NetCDF datasets.
+
+You can select as many datasets as you'd like. Once you've selected all your datasets, click `USE THESE DATASETS` on the bottom of the page to move to the next step.
+
+![Select NetCDFs](imgs/data-modeling-select-netcdfs.png)
+
+## Building a graph
+
+On the second step, you now have to start building out your graph. Drag and drop the different nodes referenced below from the list of nodes on the right side of the screen onto the dotted surface that covers the majority of the screen. Click on the small circles at the bottom/top of each node, then move your mouse (with the line from the circle) to the circle at the bottom/top of the node you'd like to connect your selected node to. By adding nodes, described below, and connecting them together, you'll create a graph that will allow you to process data from multiple features into new dataset outputs.
+
+### Nodes:
+
+![Node List](imgs/data-modeling-node-list.png)
+
+#### Load Node
+
+![Load Node](imgs/data-modeling-load-node.png)
+This starts with selecting Load Nodes. Each Load Node represents one feature from a dataset. You select this feature with the "Data Source" dropdown at the top of the node. You'll see features listed per dataset in this dropdown.
+
+Once you've selected a feature for a Load Node, you can choose to make it the Geo Resolution and/or the Time Resolution for your new dataset. You can only select each of these once in the entire graph, either in the Load Nodes or in the Time/Geo Resolution selectors in the sidebar.
+
+You also need to choose a Geo and Time Aggregation function for each Load Node.
+
+<!-- Note: explain these:
+'conserve', 'min', 'max', 'mean', 'median', 'mode', 'interp_or_mean', 'nearest_or_mode',
+ -->
+
+#### Threshold Node
+
+#### Join Node
```

Comment:
```suggestion
#### Join Node
![Join Node](https://github.com/dojo-modeling/dojo-docs/assets/41205753/ac855483-9e0a-44de-9620-58efde013bd3)
Join nodes are used to combine multiple nodes in the graph into a single dataset. Join nodes take as input two parent nodes, and outputs the values of those input nodes combined together. Under the hood, a join node multiplies the input datasets together. 

The most common use case for join nodes is to filter data to where some condition is true. A dataset mask can be created with a [[#Threshold Node]], and then that mask can be "joined" with the data that you want to apply the mask to. 

Less commonly, depending on the particular data operations you are trying to do, it may make sense to use a Join Node to multiple two separate datasets together to get a new dataset out the other end. E.g. if one dataset is gridded solar energy flux (watts/meter^2), and another dataset is solar panel area (meter^2) per each lat/lon point, you could multiply them together with a Join Node to get a new dataset that is the expected power production per each lat/lon grid point (assuming 100% efficiency). 

```